server/schedule/region_scatterer.go

```diff
@@ -29,6 +30,29 @@ import (
 
 const regionScatterName = "region-scatter"
 
+type selectedLeaderStores struct {
+	mu     sync.Mutex
+	stores map[uint64]uint64
+}
+
+func (s *selectedLeaderStores) put(id uint64) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.stores[id] = s.stores[id] + 1
+}
```

Comment:
If the leader is transferred manually, will the old store's count minus 1 and new store's count plus 1?

Reply:
Yes, you are right. This scheduler assumes that the leader and region will not change significantly after scheduling. Maybe we need to discuss optimization in another issue.

Reply:
I think we can directly to make a mechanism (syncer or tracker) to record the correct leader count distrubution.

Reply:
In fact, it will be more complicated to do this, such as `truncate table` we need to remove the regions from the tracker, and what if `recover a table`? The current approach is at least feasible in general, there will be no more operator after cluster in balanced, especially in big cluster.

Reply:
But the tracker you mentioned is a good idea, and I even want to use it to report why there is this operator produce.
cc @Yisaer